README.md

```diff
@@ -11,6 +11,65 @@ part of the array each recursive call considers.
 
 ## Runtime Analysis
 
-Analyse the time complexity of your implementation and give a $\Theta$ bound for
+Analyze the time complexity of your implementation and give a $\Theta$ bound for
 its worst-case runtime. Add your answer, including your reasoning, to this
 markdown file.
+
+Complexity Analysis: 
+Here is my code for the activity. 
+function mergesort(array) {
+    let n = array.length;
+
+    // Start the subarrays of 1 and double with each iteration 
+    for(let length = 1; length < n; length *= 2){
+        for(let left = 0; left < n - 1; left += 2 * length) {
+            let mid = Math.min(left + length - 1, n - 1); 
+            let right = Math.min(left + 2 * length - 1, n-1); 
+            merge(array, left, mid, right);
+        }
+    }
+    
+    return array;
+}
+
+function merge(array, left, mid, right) {
+    let start = left; 
+    let end = mid + 1; 
+
+    while (start <= mid && end <= right) {
+        if (array[start] <= array[end]) {
+            start++; 
+        } else {
+            let value = array[end]; 
+            let index = end; 
+
+            while (index !== start) {
+                array[index] = array[index - 1]; 
+                index--; 
+            }
+
+            array[start] = value; 
+
+            start++;
+            mid++;
+            end++;
+        }
+    }
+}
+
+Looking back at my outer loop function the length doubles with every iteration done. This will execute log_(2)n times in the runtime analysis. With that execution, it will create _O_(log n) for all the iterations done. 
+
+Since the inner loop is shifting the elements around the run time should be _O_(n). This is concerning n. This is from the processing of individual elements and the moving and splitting it is doing. 
+
+The standard runtime comes from the (n), which is the sorting or merging of elements. The log n comes from the splitting of the arrays. Which will create _O_(n log n) this is also the standard runtime for mergesort. 
+
+The standard time for the runtime with mergesort is _O_(n log n). With the inner loop being _O_(n). We have to merge them. Which will look like _O_(n x n log n) = _O_(n^2 log n). 
```

Comment:
But the inner and outer loops combined are doing the same thing as the recursive mergesort, aren't they? Then why are the complexities different?